Command to print to STDOUT

jcjgraf opened the pull request:
Adds the new command `print-stdout` to print to STDOUT from within vimiv, and an alias to print all marked images.

~Implements #379~

karlch commented:
While I see the use of a command that prints the marked images, I would prefer to keep this more general, maybe something like:
```python
@api.commands.register()
def needs_name(text: List[str], sep: str = "\n", end: str = "\n"):
    """Print text to the terminal."""
    print(*text, sep=sep, end=end)
```
which could print the marked  images just by calling `:needs-name %m`. If we like, we could even add an alias for printing the marked images.

jcjgraf commented:
I this is a great generalisation idea. I have improved the command as you have suggested and also added an alias for printing the marked images. I have also added a series of end2end test cases.

I have named the command `:print-stdout`. If this is too long or something, I would go for `:print-out`.

The alias is named `:print-mark`, tho I am unsure which of `:print-mark` and `:mark-print` is better suited. The advantage of the later is that it is more nicely integrated into the mark commands.

Let me know if you spot anything to add or improve.

karlch commented on tests/end2end/features/api/print.feature:
```diff
@@ -0,0 +1,42 @@
+        Then stdout should contain 'test1'
+        And stdout should contain 'test2'
+
+    Scenario: Print list using seq
```
seq -> sep :blush:

jcjgraf replied:
fixed in b039b82bcb453c33673e671b3d9d2ef7c306f7fa

karlch commented on tests/end2end/features/api/print.feature:
```diff
@@ -0,0 +1,42 @@
+    Scenario: Print longer list
+        Given I start vimiv
+        And I capture output
+        When I run print-stdout "test1", "test2"
```
I am not sure why there is a `,` between the two, but it doesn't do any harm of course :smile:

jcjgraf replied:
you're right. That should not be there. Fixed in b039b82bcb453c33673e671b3d9d2ef7c306f7fa

karlch commented on vimiv/api/__init__.py:
```diff
@@ -93,3 +93,19 @@ def add_external_format(
     """
     files.add_image_format(file_format, test_func)
     imagereader.external_handler[file_format] = load_func
+
```
Sorry for being a bit of a nuisance here, but personally I would not expose a command that simply wraps the builtin print function via the api. Instead this could just go into `api._modules` as is :blush:

jcjgraf replied:
I thought that if we already have a *generalized* printer, everyone should also use it. But I also do not mind changing it.

Fixed in b039b82bcb453c33673e671b3d9d2ef7c306f7fa

karlch replied:
Thanks a lot, this looks great now :blush:

karlch commented on vimiv/gui/metadatawidget.py:
```diff
@@ -123,7 +123,7 @@ def metadata_list_keys(self, n_cols: int = 3, to_term: bool = False):
             keys = sorted(set(self.handler.get_keys()))
             if to_term:
-                print(*keys, sep="\n")
+                api.print_stdout(keys)
```
As I would not expose the command as mentioned before, this would just have to be redone...

karlch reviewed (commented):
Thanks a lot, the tests look great! The only issue I have is the one concerning the exposure of the command via the api. Let me know if you have a different opinion here :blush:
